[DONTMERGE - WIP] First attempt to create DocBook 5 MAIN

tomschr opened the pull request:
* Add `chunker.xsl` from DocBook XSL
* Create `map-to-db5MAIN.xsl` to create DocBook5 output for MAIN file
* Implement previous stylesheet in `ditatodocbook.sh` script

tomschr committed:
Change upstream stylesheet to support DocBook5

sknorr commented on map-to-db5MAIN.xsl:
```diff
@@ -0,0 +1,222 @@
+<?xml version="1.0"?>
```
Please just change the existing map-to-MAIN to work with DB5 instead. I don't see much use in converting to DocBook 4 when we can have 5. (I.e. I think Dietrich if can create DB5, there is no DB4 support needed.)

sknorr commented on ditatodocbook.sh:
```diff
@@ -138,13 +144,20 @@ mkdir -p "$outputxmldir"
+  # 2> "$tmpdir/includes"
+
+# Test, if the includes file is there, otherwise abort
+[[ -e "$tmpdir/includes" ]] || die "ERROR: File $tmpdir/includes not found"
```
Use `-f` here -- I think `-e` would allow directories too

tomschr replied:
Yes, good idea! 👍

sknorr commented on ditatodocbook.sh:
```diff
@@ -85,6 +90,7 @@ CLEANTEMP=1
 CLEANID=0
 TWEAK=""
 ENTITYFILE="entities.xml"
+MAP2MAIN_XSLT="$mydir/map-to-MAIN.xsl"
```
What is this good for? Especially among the "configurable" options, this seems misplaced.

tomschr replied:
It's just my style. 😉  I don't want to repeat nor memorize the name of the stylesheet.

sknorr replied:
This is only used once. And it definitely should not be grouped together with the user-configurable options.

sknorr commented:
I've had a look at the new `map-to-db5MAIN` file. 

* I diffed between `map-to-MAIN` and `map-to-db5MAIN`. Getting a 600-line diff for a 250-line file is ... unexpected. Most of the diffed lines are (unwanted) whitespace changes.
* Your reworking currently partially destroys the features to remove/replace files at the top level. To make this work properly, you would need to copy/paste the relevant parts of normal-mode templates into the collect.files templates.
* It is also incomplete, there is still an xsl:message for content that needs to go into `includes` at the end left.

This is all stuff that had already been working.

If this is just about seeing error and warning messages on the command line, it seems to me that there are easier ways to get those (like just outputting the message caught in includes after the xsltproc run).
That may not be proper-est way to go about this but it would be quick.

tomschr committed:
Output DocBook 5

* Create xml-stylesheet PI
* Remove public and system identifiers of DOCTYPE declaration
* bookinfo -> info, @lang -> @xml:lang

sknorr commented on docbook/topic2db.xsl:
```diff
@@ -115,13 +122,13 @@
-      <xsl:when test="$contextType='glossary'">docinfo</xsl:when>
-      <xsl:when test="$contextType='part'">docinfo</xsl:when>
-      <xsl:when test="$contextType='section'">sectioninfo</xsl:when>
+      <xsl:when test="$contextType='article'">info</xsl:when>
```
I think this when is unnecessary now.

tomschr committed:
WIP: DocBook5 namespaces

tomschr committed:
Fix clean-blocks.xsl

* Correct dbselfblocks entity
* Exclude namespace prefix d
* Make template recursive

tomschr committed:
Replace while block with clean-blocks.xsl

Commit:
Make clean-ids.xsl fail less -- initialize namespace correctly

There is still a whole bunch of xmlns=... attributes everywhere.

Also remove @id here, since it is now useless

Commit:
Fix indentation

sknorr commented:
Hm, there still seem to be a few issues here that I didn't figure out:

* we're somehow writing the DB5 namespaces in lots of elements, not just the root element
* daps xmlformat does not run, the error is `ERROR: Failed to get filelist for ROOTID ""`

Also, I have moved your map/bookmap commit onto https://github.com/sknorr/dietrich/tree/f/map-to-bookmap

Commit:
Completely remove commented code in DITA->DB stylesheets

Commit:
Fix indentation of PI

Commit:
Horrible workarounds but DB5 now actually works
